packages/typespec-test/test/anomalyDetector/generated/typespec-ts/src/models.ts

```diff
@@ -2,14 +2,14 @@
 // Licensed under the MIT license.
 
 /** The request of entire or last anomaly detection. */
-export interface UnivariateDetectionOptions {
+export interface UnivariateUnivariateDetectionOptions {
```

Comment:
This is the change we get if we by default add the namespace as prefix for models.

Reply:
Do you know how expensive it would be to implement this only when there are conflicts? If it is not too expensive we can go that route and have a flag in case we want to turn it on/off

Reply:
Add an option  enableModelNamespace to allow customer to append the namespace information in the model name.  By default, we will detect if there's model name conflict, if yes we will set this option as true when this option is unset. But if customer has specifically set a true or false value to it, we will just respect their choice. 
Also, this option applies to both RLC and Modular.

Reply:
In the anomalyDetector's case, we have enabled this enableModelNamespace: true, it will use the namespace in the prefix, even if it doesn't have conflicts.
In the hierarchy generic's case, we don't set this option, but it detect there's a model name conflict, so it will enable it to add the full namespace prefix in the model name.